lld/ELF/Driver.cpp

```diff
@@ -2617,12 +2624,28 @@ static void checkAndReportMissingFeature(StringRef config, uint32_t features,
 //
 // This is also the case with AARCH64's BTI and PAC which use the similar
 // GNU_PROPERTY_AARCH64_FEATURE_1_AND mechanism.
-static uint32_t getAndFeatures() {
+//
+// For AArch64 PAuth-enabled object files, the core info of all of them must
+// match. Missing info for some object files with matching info for remaining
+// ones can be allowed (see -z pauth-report).
+static void readSecurityNotes() {
   if (config->emachine != EM_386 && config->emachine != EM_X86_64 &&
       config->emachine != EM_AARCH64)
-    return 0;
+    return;
+
+  config->andFeatures = -1;
+
+  StringRef referenceFileName;
```

Comment:
This loop tries to find a reference filename, but the loop can be removed.

Consider cherry picking the simplification at https://github.com/maskray/llvm-project/tree/lld-pauth ?

Reply:
You are right, and thanks for the test improvement. 

Add single quotes around `referenceFileName` and this patch LGTM!